src/config.ts

```diff
@@ -0,0 +1,65 @@
+const hasDOM = typeof document !== 'undefined' && typeof window !== 'undefined';
+const scriptElement: HTMLScriptElement = hasDOM && document.currentScript;
+// @ts-ignore
+const globalConfig: Record<String, string | string[] | null> =
+	globalThis.Prism?.constructor?.name === 'Object' ? globalThis.Prism : {};
+
+function getGlobalSetting (name: string) {
+	let camelCaseName = name.replace(/-([a-z])/g, g => g[1].toUpperCase());
+
+	if (camelCaseName in globalConfig) {
+		return globalConfig[camelCaseName];
+	}
+	else if (name in globalConfig) {
+		return globalConfig[name];
+	}
+	else if (hasDOM) {
+		return (
+			scriptElement.dataset[camelCaseName] ??
```

Comment:
```suggestion
			scriptElement!.dataset[camelCaseName] ??
```

If I'm not mistaken, `scriptElement` will be defined here, but TS can't figure it out from the code, so we need to tell it we're sure with what we do.